include/lbann/layers/io/input/generic_input_layer.hpp

```diff
@@ -848,21 +928,26 @@ class generic_input_layer : public io_layer {
   }
 
  protected:
+  std::vector<generic_io_buffer*> m_io_buffers;
```

Comment:
I would like to defer making this change until I clean up the rest of the I/O system and get rid of the partitioned and distributed I/O buffers.  At that point, it will be easier to change this to a unique pointer.